barretenberg/cpp/src/barretenberg/common/constexpr_utils.hpp

```diff
@@ -4,16 +4,22 @@
 #include <tuple>
 #include <utility>
 
+namespace bb {
+namespace detail {
+
 /**
- * @brief constexpr_utils defines some helper methods that perform some stl-equivalent operations
- * but in a constexpr context over quantities known at compile-time
- *
- * Current methods are:
- *
- * constexpr_for : loop over a range , where the size_t iterator `i` is a constexpr variable
- * constexpr_find : find if an element is in an array
+ * @brief Create an index sequence from Min to Max (not included) with an increment of Inc
  */
-namespace bb {
+template <size_t Min, size_t Max, size_t Inc> constexpr auto make_index_range()
+{
+    static_assert(Max >= Min);
+    static_assert(Inc >= 1);
+    return []<size_t... Is>(std::index_sequence<Is...>) {
```

Comment:
Should I add tests for this? Is not super self-evident.

Reply:
I can also just leave the original one for this, since the recursion didn't seem to be hurting.